src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -1,180 +1,165 @@
+  const venueEvents = useSelector(venueEventsSelector) ?? emptyEventsArray;
   const { worldUsers } = useWorldUsers();
 
   const { openUserProfileModal } = useProfileModalControls();
 
-  useEffect(() => {
-    const normalizedSearchQuery = searchQuery.toLowerCase();
-    if (!normalizedSearchQuery) {
-      setSearchResult({
-        rooms: [],
-        users: [],
-        events: [],
-      });
-      return;
+  const foundRooms = useMemo<JSX.Element[]>(() => {
+    if (!searchQuery) {
+      return [];
     }
 
-    const venueUsersResults = worldUsers.filter((user) =>
-      user.partyName?.toLowerCase()?.includes(normalizedSearchQuery)
+    /* @debt we really shouldn't be using the index as part of the key here, it's unstable.. but rooms don't have a unique identifier */
+    return (
+      venue?.rooms
+        ?.filter(
+          (room) =>
+            room.title.toLowerCase().includes(searchQuery) &&
+            room.type !== RoomTypes.unclickable
+        )
+        .map((room, index) => (
+          <NavSearchResult
+            key={`room-${room.title}-${index}`}
+            title={room.title}
+            description="Room"
+            image={room.image_url}
+            onClick={() => {
+              setSelectedRoom(room);
+              clearSearchQuery();
+            }}
+          />
+        )) ?? []
     );
+  }, [searchQuery, venue, clearSearchQuery]);
 
-    const venueEventsResults = venueEvents.filter((event) =>
-      event.name.toLowerCase().includes(normalizedSearchQuery)
-    );
+  const foundUsers = useMemo<JSX.Element[]>(() => {
+    if (!searchQuery) {
+      return [];
+    }
 
-    const roomsResults: Room[] =
-      venue?.rooms?.filter(
-        (room) =>
-          room.title.toLowerCase().includes(normalizedSearchQuery) &&
-          room.type !== RoomTypes.unclickable
-      ) ?? [];
+    return worldUsers
+      .filter((user) => user.partyName?.toLowerCase().includes(searchQuery))
+      .map((user) => (
+        <NavSearchResult
+          key={`user-${user.id}`}
+          title={user.partyName as string}
+          image={user.pictureUrl}
+          isAvatar={true}
+          onClick={() => {
+            openUserProfileModal(user);
+            clearSearchQuery();
+          }}
+        />
+      ));
+  }, [searchQuery, worldUsers, clearSearchQuery, openUserProfileModal]);
+
+  const foundEvents = useMemo<JSX.Element[]>(() => {
+    if (!searchQuery) {
+      return [];
+    }
 
-    setSearchResult({
-      rooms: roomsResults,
-      users: venueUsersResults,
-      events: venueEventsResults,
-    });
-  }, [searchQuery, venue, venueEvents, worldUsers]);
+    return venueEvents
+      .filter((event) => event.name.toLowerCase().includes(searchQuery))
+      .map((event) => {
+        // @debt event can be related to the current venue as well but we don't receive its icon
+        const room = venue?.rooms?.find((room) => room.title === event.room);
```

Comment:
I believe this `.find` can cause `room` to be potentially `undefined` yeah?

We might want to handle that case more explicitly below? Eg. Where we are using `room?.image_url`?

Could this be related to why the `@debt` comment above is happening?

Reply:
see the comment above, please. Let's move the discussion about it there. https://github.com/sparkletown/sparkle/pull/1328#discussion_r617636340